File: Corpo.Domain/Models/WeeklyTemplate.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Corpo.Domain.Models
{
    public class WeeklyTemplate
    {

        public int Id { get; set; }
        public string Name { get; set; }
        public string Goal { get; set; }
        public List<WeeklyWodTemplate> WeeklyWodTemplates { get; set; }
    }
}

File: Corpo.Web/ClientApp/src/app/domain/wod/weekly-template.ts
import { WodTemplate } from "../wod";
import { WeeklyWodTemplate } from "./weekly-wod-template";

export class WeeklyTemplate {
  id: number;
  name: string;
  goal: string;
  weeklyWodTemplates: WeeklyWodTemplate[] = [];
 
}

export class WeeklyTemplateResponse {
  id: number;
  name: string;
  goal: string;
  wodTemplates: WodTemplate[] = [];
}
